### PR TITLE
Features/include reward in economy

### PR DESCRIPTION
**Pull request by nowooj**
- using calculation total supply
- divide inflation & fare
- include fare in distribute

### CODE REVIEW

File: execution-engine/contracts/hdac-system/pop-install/src/lib.rs [hunk 1]
```diff
@@ -26,13 +26,13 @@ const POS_REWARDS_PURSE: &str = "pos_rewards_purse";
...
 const POS_COMMUNITY_PURSE: &str = "pos_community_purse";
 const POS_FUNCTION_NAME: &str = "pos_ext";
-const BIGSUN_TO_HDAC: u64 = 1_000_000_000_000_000_000_u64;
 
 #[repr(u32)]
 enum Args {
     MintURef = 0,
     GenesisValidators = 1,
     StateInformations = 2,
+    TotalAvaliableAmount = 3,
```
Comment (psy2848048) [inline execution-engine/contracts/hdac-system/pop-install/src/lib.rs:35]: `TotalAvailableAmount`

File: execution-engine/contracts/hdac-system/pop-install/src/lib.rs [hunk 2]
```diff
@@ -192,26 +225,47 @@ pub extern "C" fn call() {
...
+        .unwrap_or_revert_with(ApiError::MissingArgument)
+        .unwrap_or_revert_with(ApiError::InvalidArgument);
+
     // Insert total supply
+    let total_supply = total_amount
+        + total_bonds
+        + total_inflation_commission
+        + total_inflation_reward
+        + total_fare_commission
+        + total_fare_reward;
```
Comment (psy2848048) [inline execution-engine/contracts/hdac-system/pop-install/src/lib.rs:244]: Fare seems better to be distributed with only validator or current BP

File: execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs [hunk 3]
```diff
@@ -228,20 +243,31 @@ impl ProofOfProfessionContract {
...
+        let mut inflation_rewards = ContractClaim::read_reward(INFLATION_REWARD)?;
+        let user_inflation_reward = inflation_rewards
+            .0
+            .get(user)
+            .cloned()
+            .unwrap_or_revert_with(Error::RewardNotFound);
+        inflation_rewards.claim_rewards(user, &user_inflation_reward);
+        ContractClaim::write_reward(INFLATION_REWARD, &inflation_rewards);
+
+        let mut fare_rewards = ContractClaim::read_reward(FARE_REWARD)?;
```
Comment (psy2848048) [inline execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs:255]: But recommend not to delete, just block them, and a developer far far later can be used.

**Comment by nowooj** [comment]
I will do this after applying https://github.com/hdac-io/CasperLabs/pull/33.

**Pull request closed without merging**